cmd/utils/flags.go

```diff
@@ -2158,3 +2184,26 @@ func MakeConsolePreloads(ctx *cli.Context) []string {
 	}
 	return preloads
 }
+
+// ParseStateScheme resolves scheme identifier from CLI flag.
+func ParseStateScheme(ctx *cli.Context) string {
+	if ctx.Bool(PathBasedSchemeFlag.Name) {
+		return rawdb.PathScheme
+	}
+	return rawdb.HashScheme
+}
+
+// MakeDatabase constructs a trie database based on the configured scheme.
+func MakeDatabase(ctx *cli.Context, disk ethdb.Database, readOnly bool) *trie.Database {
```

Comment:
Perhaps rename to MakeTrieDatabase, otherwise there's no context to figure out what we're trying to do via a utils package.